### DISCUSSION TITLE
Is there a way to not escape PathParams?

### DISCUSSION THREAD

**Discussion by SVilgelm** [go-resty/resty#570] [Q&A] [answered]
Hi, as it is said in the subject. I want to use the PathParams without escaping. Is there anyway to do that?

**Comment by kecci** [comment]
Hi @SVilgelm ,

I look into the docs, we can `SetPathParams` and the default is not escaped.

For example:
```go
package main

import "github.com/go-resty/resty/v2"

func main() {
	// Create a Resty Client
	client := resty.New()

	client.R().SetPathParams(map[string]string{
		"userId":       "sample@sample.com",
		"subAccountId": "100002",
	}).
		Get("https://webhook.site/9e0d8a64-0ed1-4e28-ae32-ddd26fae5e1d/{userId}/{subAccountId}")

	// Result:
	//   Composed URL - https://webhook.site/9e0d8a64-0ed1-4e28-ae32-ddd26fae5e1d/sample@sample.com/100002
}
```

Output with webhook.site :
<img width="1512" alt="Screen Shot 2022-10-07 at 14 02 54" src="https://user-images.githubusercontent.com/25241073/194487756-6f16da49-b800-43ae-bed7-271dd5f34e33.png">

You can see the docs here: https://github.com/go-resty/resty#request-url-path-params

Hope this help answered your questions.

**Comment by SVilgelm** [comment]
Here is my example:
```go
package main

import (
        "fmt"

        "github.com/go-resty/resty/v2"
)

func main() {
        resp, err := resty.New().R().SetPathParam("id", "foo/bar/xyz").Get("https://webhook.site/44923f70-4cb5-464c-ac76-217762c5cee8/{id}")
        fmt.Printf("err: %s\n", err)
        fmt.Printf("resp: %s: %s\n", resp.Status(), resp.String())
        fmt.Printf("url: %s", resp.Request.URL)
}
```

output:
```shell
% go run main.go
err: %!s(<nil>)
resp: 200 OK: 
url: https://webhook.site/44923f70-4cb5-464c-ac76-217762c5cee8/foo%2Fbar%2Fxyz%   
```

https://webhook.site/#!/44923f70-4cb5-464c-ac76-217762c5cee8/0f62631f-e8d4-4799-b9a7-5b353154b1fa/1
<img width="646" alt="Screen Shot 2022-10-07 at 1 34 00 PM" src="https://user-images.githubusercontent.com/523825/194648326-6cd8f533-9ed4-40e7-ae87-b8c9d49cb082.png">

as you can see the value of `id` parameter was escaped

**Comment by jeevatkm** [comment, marked as answer]
The RawPathParams feature was added in https://github.com/go-resty/resty/pull/664 and released with v2.8.0. Of course, you know it.
